packages/ODEs/src/app.ts

```diff
@@ -1,654 +1,870 @@
-  {label: `${CONTROL_EXPR.INITS}:\n  `, type: "keyword", info: INFO.INITS},
-  {label: `${CONTROL_EXPR.PARAMS}:\n  `, type: "keyword", info: INFO.PARAMS},
-  {label: `${CONTROL_EXPR.CONSTS}:\n  `, type: "keyword", info: INFO.CONSTS},
-  {label: `${CONTROL_EXPR.TOL}: `, type: "keyword", info: INFO.TOL},
-  {label: `${CONTROL_EXPR.LOOP}:\n  `, type: "keyword", info: INFO.LOOP},
-  {label: `${CONTROL_EXPR.UPDATE}:\n  `, type: "keyword", info: INFO.UPDATE},
-  {label: `${CONTROL_EXPR.OUTPUT}:\n  `, type: "keyword", info: INFO.OUTPUT},
-  {label: `${CONTROL_EXPR.COMMENT}: `, type: "keyword", info: INFO.COMMENT},
+  {label: `${CONTROL_EXPR.NAME}: `, type: 'keyword', info: INFO.NAME},
+  {label: `${CONTROL_EXPR.TAGS}: `, type: 'keyword', info: INFO.TAGS},
+  {label: `${CONTROL_EXPR.DESCR}: `, type: 'keyword', info: INFO.DESCR},
+  {label: `${CONTROL_EXPR.DIF_EQ}:\n  `, type: 'keyword', info: INFO.DIF_EQ},
+  {label: `${CONTROL_EXPR.EXPR}:\n  `, type: 'keyword', info: INFO.EXPR},
+  {label: `${CONTROL_EXPR.ARG}: `, type: 'keyword', info: INFO.ARG},
+  {label: `${CONTROL_EXPR.INITS}:\n  `, type: 'keyword', info: INFO.INITS},
+  {label: `${CONTROL_EXPR.PARAMS}:\n  `, type: 'keyword', info: INFO.PARAMS},
+  {label: `${CONTROL_EXPR.CONSTS}:\n  `, type: 'keyword', info: INFO.CONSTS},
+  {label: `${CONTROL_EXPR.TOL}: `, type: 'keyword', info: INFO.TOL},
+  {label: `${CONTROL_EXPR.LOOP}:\n  `, type: 'keyword', info: INFO.LOOP},
+  {label: `${CONTROL_EXPR.UPDATE}:  `, type: 'keyword', info: INFO.UPDATE},
+  {label: `${CONTROL_EXPR.OUTPUT}:\n  `, type: 'keyword', info: INFO.OUTPUT},
+  {label: `${CONTROL_EXPR.COMMENT}: `, type: 'keyword', info: INFO.COMMENT},
 ];
 
 /** Control expressions completion utilite */
 function contrCompletions(context: any) {
-  let before = context.matchBefore(/[#]\w*/)
-  if (!context.explicit && !before) return null
+  const before = context.matchBefore(/[#]\w*/);
+
+  if (!context.explicit && !before) return null;
   return {
     from: before ? before.from : context.pos,
     options: completions,
-    validFor: /^\w*$/
-  }
+    validFor: /^\w*$/,
+  };
 }
 
 /** Return options of line chart */
-function lineChartOptions(colNames: string[]): Object {
+function getLineChartOptions(colNames: string[]): Object {
   const count = colNames.length;
-
   return {
     xColumnName: colNames[0],
-    yColumnNames: (count > 1) ? colNames.slice(1) : colNames[0],     
+    yColumnNames: (count > 1) ? colNames.slice(1).filter((name) => name !== STAGE_COL_NAME) : colNames[0],
     showTitle: true,
-    sharex: true, 
+    sharex: true,
     multiAxis: count > MAX_LINE_CHART,
-    multiAxisLegendPosition: "RightTop",
+    multiAxisLegendPosition: 'RightTop',
+    segmentColumnName: colNames.includes(STAGE_COL_NAME) ? STAGE_COL_NAME: null,
   };
 }
 
-/** Run solver application */
-export async function runSolverApp() {
+/**  String to value */
+const strToVal = (s: string) => {
+  const num = Number(s);
```

Comment:
to convert a number to a string you can simply use const num = +s;

Reply:
const num = +s;

Ouch. I would have never guessed what it does. I would suggest to probably avoid it for readability.